fix: Show title, inputs, and outputs on Vue node component even when node fails to initalize (e.g., if missing)

christian-byrne opened the pull request:
## Summary

Enhanced missing node title resolution with improved fallback logic and proper data extraction timing during graph configuration.

## Changes

- **What**: Improved title fallback chain in `useGraphNodeManager` (`title → type → constructor.comfyClass → constructor.title → constructor.name → 'Unknown'`)
- **What**: Added re-extraction of Vue node data after `onAfterGraphConfigured` to capture populated title/slots/widgets
- **What**: Enhanced `NodeHeader` title resolution with proper TypeScript typing and descriptive variable names

## Review Focus

Title resolution during workflow loading vs immediate initialization and fallback behavior for truly missing nodes.

## Screenshots (if applicable)

<img width="1404" height="1140" alt="Selection_2163" src="https://github.com/user-attachments/assets/bd20b90b-aff1-4d76-ab3a-3de9588fdfbf" />

┆Issue is synchronized with this [Notion page](https://www.notion.so/PR-5540-fix-Show-title-inputs-and-outputs-on-Vue-node-component-even-when-node-fails-to-inital-26d6d73d3650819a962bcaf53194a63f) by [Unito](https://www.unito.io)

christian-byrne committed:
add missing node error border

christian-byrne committed:
update vue node data after configure

DrJKL commented on src/composables/graph/useGraphNodeManager.ts:
```diff
@@ -204,8 +204,13 @@ export const useGraphNodeManager = (graph: LGraph): GraphNodeManager => {
-      title: node.title || 'Untitled',
-      type: node.type || 'Unknown',
+      title: typeof node.title === 'string' ? node.title : '',
+      type:
+        node.type ||
+        node.constructor?.comfyClass ||
+        node.constructor?.title ||
+        node.constructor?.name ||
+        'Unknown',
```
With this many options, would you mind pulling it out into a named const, for readability?

DrJKL commented on src/renderer/extensions/vueNodes/components/NodeHeader.vue:
```diff
@@ -74,15 +74,23 @@ const isEditing = ref(false)
+  () => {
+    const next = resolveTitle(nodeInfo.value)
+    if (next !== displayTitle.value) {
+      displayTitle.value = next
```
Is this a computed?

christian-byrne replied:
Heh, fixed https://github.com/Comfy-Org/ComfyUI_frontend/pull/5540/commits/fed177eaf4ca013c67b058021684e347a477961c

DrJKL commented on src/renderer/extensions/vueNodes/components/NodeHeader.vue:
```diff
@@ -74,15 +74,23 @@ const isEditing = ref(false)
+  const type = (info?.type ?? '').trim()
+  return type.length > 0 ? type : 'Untitled'
+}
+
```
Are these worth extracting?

christian-byrne replied:
The strings? Sure https://github.com/Comfy-Org/ComfyUI_frontend/pull/5540/commits/fed177eaf4ca013c67b058021684e347a477961c

DrJKL replied:
Oh, I meant the function, since it's pure and feels utils-like.

DrJKL reviewed (dismissed):
I think there's one computed that's pretending to be a watch, and some nits about structure, but 👍🏻

christian-byrne committed:
Merge branch 'main' into vue-nodes/feat/skeleton-missing-nodes

christian-byrne committed:
[refactor] extract node type resolution to named const - addresses @DrJKL's readability concern

Extracted the multi-fallback type resolution logic into a clearly named
variable for improved readability and maintainability.

christian-byrne committed:
[refactor] convert watch to computed pattern - addresses @DrJKL's structure comment

Replaced ref + watch pattern with computed for displayTitle, providing cleaner
reactive behavior and eliminating the need for manual sync logic.
